tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_SWT.java

```diff
@@ -117,6 +119,51 @@ public void test_getMessageLjava_lang_String() {
 
 }
 
+private String pathFromClass(Class classValue) {
+	String filePrefix = "file:/";
+
+	String url = classValue.getProtectionDomain().getCodeSource().getLocation().toString();
+	assertTrue(url.startsWith(filePrefix));
+
+	String urlPath = url.substring(filePrefix.length());
+	String path = Paths.get(urlPath).toAbsolutePath().toString();
+
+	// On Windows, use / for consistency
+	return path.replace('\\', '/');
+}
+
+/**
+ * Ensure that SWT being tested was built together with the tests. This
+ * is a test designed for build scripts, such as the one that checks
+ * GitHub Pull Requests.
+ */
+@Test
+public void test_isLocal() {
+	// If you change default to NO, make sure that this test runs on GitHub
+	if (Boolean.getBoolean("org.eclipse.swt.tests.junit.disable.test_isLocal")) {
+		return;
+	}
+
+	String swtPath = pathFromClass(SWT.class);
+	String tstPath = pathFromClass(Test_org_eclipse_swt_SWT.class);
+
+	// It's hard to devise a condition that would hold for all possible
+	// locations of compiled classes. For example, my own project uses
+	// output directories different from what Eclipse does by default.
+
+	// One thing for sure, SWT shouldn't be on maven cache!
+	if (swtPath.contains("/.m2/")) {
```

Comment:
I think this needs be more specific like 
```suggestion
	if (swtPath.contains("/.m2/repository/p2/osgi/bundle/")) {
```

Reply:
`/.m2/repository` part is probably good, and I'm not sure about the rest. I think it's best to just leave it as is for now, and if it causes false positives, it can be improved later when it's clear when exactly this could happen.

Reply:
No, it is fine that the jar comes from the m2 repository, but it should not come from P2 (p2 artifacts are stored under` p2/osgi/bundle/`).

> if it causes false positives

It already does, see the Jenkins-job where it actually is at the expected location:

/home/vnc/.m2/repository/org/eclipse/swt/org.eclipse.swt.gtk.linux.x86_64/3.121.0-SNAPSHOT/org.eclipse.swt.gtk.linux.x86_64-3.121.0-SNAPSHOT.jar

Reply:
Thanks for clarification. I have almost zero maven experience, and two days ago, it was zero exactly :)